src/Advection/weno_smoothness.jl

```diff
@@ -0,0 +1,201 @@
+const SS20M = (0, 0, 0)
+const SS210 = (1, 0, 0)
+const SS30M = (0, 0, 0, 0, 0, 0)
+const SS310 = (1, 0, 0, 0, 0, 0)
+const SS40M = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+const SS410 = (1, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+const SS50M = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+const SS510 = (1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+
+# Coefficients for third order WENO
+const SS220 = (1, -2, 1)
+const SS221 = (1, -2, 1)
+
+const SS320 = (1, -2, 0, 1, 0, 0)
+const SS321 = (1, -2, 0, 1, 0, 0)
+
+const SS420 = (1, -2, 0, 0, 1, 0, 0, 0, 0, 0)
+const SS421 = (1, -2, 0, 0, 1, 0, 0, 0, 0, 0)
+
+const SS520 = (1, -2, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+const SS521 = (1, -2, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+
+# Coefficients for fifth order WENO
+const SS330 = (10, -31, 11, 25, -19, 4)
+const SS331 = (4,  -13,  5, 13, -13, 4)
+const SS332 = (4,  -19, 11, 25, -31, 10)
+
+const SS430 = (10, -31, 11, 0, 25, -19, 0,  4, 0, 0)
+const SS431 = (4,  -13,  5, 0, 13, -13, 0,  4, 0, 0)
+const SS432 = (4,  -19, 11, 0, 25, -31, 0, 10, 0, 0)
+
+const SS530 = (10, -31, 11, 0, 0, 25, -19, 0, 0,  4, 0, 0, 0, 0, 0)
+const SS531 = (4,  -13,  5, 0, 0, 13, -13, 0, 0,  4, 0, 0, 0, 0, 0)
+const SS532 = (4,  -19, 11, 0, 0, 25, -31, 0, 0, 10, 0, 0, 0, 0, 0)
+
+# Coefficients for seventh order WENO
+const SS440 = (2.107, -9.402, 7.042, -1.854, 11.003, -17.246, 4.642,  7.043, -3.882, 0.547)
+const SS441 = (0.547, -2.522, 1.922, -0.494,  3.443, -5.966,  1.602,  2.843, -1.642, 0.267)
+const SS442 = (0.267, -1.642, 1.602, -0.494,  2.843, -5.966,  1.922,  3.443, -2.522, 0.547)
+const SS443 = (0.547, -3.882, 4.642, -1.854,  7.043, -17.246, 7.042, 11.003, -9.402, 2.107)
+
+const SS540 = (2.107, -9.402, 7.042, -1.854, 0, 11.003, -17.246, 4.642, 0,  7.043, -3.882, 0, 0.547, 0, 0)
+const SS541 = (0.547, -2.522, 1.922, -0.494, 0,  3.443, -5.966,  1.602, 0,  2.843, -1.642, 0, 0.267, 0, 0)
+const SS542 = (0.267, -1.642, 1.602, -0.494, 0,  2.843, -5.966,  1.922, 0,  3.443, -2.522, 0, 0.547, 0, 0)
+const SS543 = (0.547, -3.882, 4.642, -1.854, 0,  7.043, -17.246, 7.042, 0, 11.003, -9.402, 0, 2.107, 0, 0)
+
+# Coefficients for ninth order WENO
+const SS550 = (1.07918,  -6.49501, 7.58823, -4.11487,  0.86329,  10.20563, -24.62076, 13.58458, -2.88007, 15.21393, -17.04396, 3.64863,  4.82963, -2.08501, 0.22658)
+const SS551 = (0.22658,  -1.40251, 1.65153, -0.88297,  0.18079,   2.42723,  -6.11976,  3.37018, -0.70237,  4.06293,  -4.64976, 0.99213,  1.38563, -0.60871, 0.06908)
+const SS552 = (0.06908,  -0.51001, 0.67923, -0.38947,  0.08209,   1.04963,  -2.99076,  1.79098, -0.38947,  2.31153,  -2.99076, 0.67923,  1.04963, -0.51001, 0.06908)
+const SS553 = (0.06908,  -0.60871, 0.99213, -0.70237,  0.18079,   1.38563,  -4.64976,  3.37018, -0.88297,  4.06293,  -6.11976, 1.65153,  2.42723, -1.40251, 0.22658)
+const SS554 = (0.22658,  -2.08501, 3.64863, -2.88007,  0.86329,   4.82963, -17.04396, 13.58458, -4.11487, 15.21393, -24.62076, 7.58823, 10.20563, -6.49501, 1.07918)
```

Comment:
should these be FP32?

Reply:
Right, we could enforce them to be Float32 probably. I think these will not be uniformly shared between threads in a warp because now the thread itself chooses the coefficients, and they might differ from thread to thread.

Reply:
Also just as a matter of practice, it is probably best to use the lowest precision number that describes it